distributed/comm/ucx.py

```diff
@@ -316,4 +318,51 @@ def get_local_address_for(self, loc):
         return unparse_host_port(local_host, None)
 
 
+def should_peek(frame):
+    header_start = b'\x83\xa7headers'
+    peek_bytes = len(header_start)
+
+    return type(frame) == memoryview and frame[:peek_bytes] == header_start
+
+
+def peek(frame):
+    """
+    Inspect a header for whether we can take a faster recv-path.
+
+    Parameters
+    ----------
+    frame : memoryview
+        The header frame to inspect.
+
+    Returns
+    -------
+    sizes : list
+        List of the next sided :meth:`recv_obj` receives to perform.
+        The recevies should be performed last to first, so use
+        :func:`list.pop` to get the next receive.
+    gpu_inbound : int
+        The number of GPU recvies to do. Decrement this to get
+        back to regular recvs.
+
+    See Also
+    --------
+    should_peek : check whether it's appropriate to peek.
+    """
+    headers = msgpack.loads(frame, use_list=False)
```

Comment:
I recommend grepping for `msgpack.loads` to see how we usually do this.  You can avoid the bytestring keys if you want to with the `raw=False` keyword.